packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -3370,6 +3406,32 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
     _setSelection(newSelection, cause);
   }
 
+  /// Select a paragraph around the location of the last tap down.
+  ///
+  /// {@macro flutter.rendering.RenderEditable.selectPosition}
+  void selectParagraph({ required SelectionChangedCause cause }) {
```

Comment:
> Can we do this in `text_selection` instead? I'm a bit reluctant to add this as a public API, since the definition of "paragraph" here seems to be highly specific to this use case, and bundling a boundary analysis API and a side effect (selecting the text) is strange.

Yeah sure



> Also you might want to check if the text is obscured (although it's very unlikely to have hard line breaks in a password field).

Good point. For obscured text, we'd just want to select everything right?

Reply:
yes.

Reply:
@LongCatIsLooong I looked into moving it, but I'm having second thoughts on doing it in this PR.

Here are some thoughts on why:
* A lot of the logic only used in text selection is already in RenderEditable. For example, `selectWordEdge`, `selectWordsInRange`. When I do a search they are only used inside of TextSelection
* The API to arbitrarily set a selection `_setSelection` is private
* It would make triple-click implemented as a special case vs. copying what the other implementations do.

I agree with the high level idea. I see a lot of logic in RenderEditable that is specific to text selection. Perhaps we would want to add a public API for arbitrarily setting the selection and pull all text selection code out? 

That being said, I don't have a *strong* opinion on it. My priority is to improve the text selection UX for our own product, so I'm open to whatever you and others on the Flutter team prefer :)